packages/components-js/projects/partials/scripts/generateInitialStylesPartial.ts

```diff
@@ -65,16 +70,91 @@ type GetInitialStylesOptionsWithoutTags = Omit<GetInitialStylesOptions, 'format'
         fontStyle: 'normal',
       },
 
-      'a, button, input, select, textarea': {
-        outline: '1px solid transparent',
-        outlineOffset: '1px',
+      'a, button': {
+        position: 'relative',
+        outline: 0, // hide default focus outline
+        '&:focus::before': {
+          content: '""',
+          position: 'absolute',
+          borderRadius: borderRadiusMedium,
+          border: `${borderWidthBase} solid ${themeLight.state.focus}`,
+          ...getInsetJssStyle(-4),
+        },
+        '&:focus:not(:focus-visible)::before': {
+          border: 0,
+        },
+        '&:disabled': {
```

Comment:
Should become a part of `:not(:disabled)` selector above instead of overriding it afterwards.